resolwe/flow/managers/listener/plugin.py

```diff
@@ -0,0 +1,125 @@
+"""Listener plugin and plugin manager class."""
+import abc
+from typing import (
+    Callable,
+    Dict,
+    Iterator,
+    MutableMapping,
+    Optional,
+    Set,
+    Type,
+)
+
+from resolwe.flow.executors.socket_utils import (
+    Message,
+    Response,
+)
+
+
+class ListenerPlugin(metaclass=abc.ABCMeta):
+    """Base class for listener plugin classes."""
+
+    @classmethod
+    def __init_subclass__(cls: Type["ListenerPlugin"], **kwargs):
+        """Register class with the registry on initialization."""
+        super().__init_subclass__(**kwargs)
+        ListenerPlugins().add_plugin(cls)
+
+    def _get_handler_name(self, command_name: str) -> str:
+        """Get the handler name."""
+        return f"handle_{command_name}"
+
+    def get_handler(
+        self, command_name: str
+    ) -> Optional[Callable[[Message, "Processor"], Response]]:
+        """Return the plugin handler
+
+        When no such plugin exists return None.
+        """
+        return getattr(self, self._get_handler_name(command_name), None)
+
+    def get_handlers(self) -> Set[str]:
+        """Return a list of handlers."""
+        return {
+            method_name
+            for method_name in dir(self)
+            if (
+                method_name.startswith("handle_")
```

Comment:
`'handle_'` is used twice in this class, so it would be nice to add it as a constant.

Reply:
OK.